### PR TITLE
browser: remove `bottle :unneeded`

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

See #75943.

### CODE REVIEW

**Comment by carlocab** [comment]
Not cool.
```
❯ sha256sum *.tar.gz
3cc601c051723db712bdff3dded97a88dacb4a2fc620df6faa856f095e52a886  browser--7.arm64_big_sur.bottle.tar.gz
61591e3772dfe5b3e0895592dbb06800bed67803e462ef3c6c80aaa0f7792aee  browser--7.big_sur.bottle.tar.gz
c8d43fec42584df0588aaebe5a7c04fe07b2b56fae6ebe7119e83901e19a0ab3  browser--7.catalina.bottle.tar.gz
a3affb0e6fb9be7a95463f256bc04fb293b1c2f86b943ba9778bbc7270d8e890  browser--7.mojave.bottle.tar.gz
```

**Comment by Bo98** [comment]
That's a bit weird.

**Comment by carlocab** [comment]
CC @Bo98 @MikeMcQuaid 

The install method is a pure `bin.install`, so there really should be no diff here. It's all just timestamps.

**Comment by carlocab** [comment]
```
❯ diffoscope browser--7.arm64_big_sur.bottle.tar.gz browser--7.big_sur.bottle.tar.gz
--- browser--7.arm64_big_sur.bottle.tar.gz
+++ browser--7.big_sur.bottle.tar.gz
├── filetype from file(1)
│ @@ -1 +1 @@
│ -gzip compressed data, was "browser-bottle.tar", last modified: Tue Apr 27 14:49:53 2021, from Unix
│ +gzip compressed data, was "browser-bottle.tar", last modified: Tue Apr 27 16:54:40 2021, from Unix
│   --- browser--7.arm64_big_sur.bottle.tar
├── +++ browser--7.big_sur.bottle.tar
│ ├── file list
│ │ @@ -1,5 +1,5 @@
│ │ -drwxr-xr-x   0        0        0        0 2021-04-27 14:49:53.000000 browser/7/
│ │ -drwxr-xr-x   0        0        0        0 2021-04-27 14:49:53.000000 browser/7/.brew/
│ │ --rw-r--r--   0        0        0      421 2021-04-27 14:49:53.000000 browser/7/.brew/browser.rb
│ │ -drwxr-xr-x   0        0        0        0 2021-04-27 14:49:53.000000 browser/7/bin/
│ │ --r-xr-xr-x   0        0        0      419 2021-04-27 14:49:53.000000 browser/7/bin/browser
│ │ +drwxr-xr-x   0        0        0        0 2021-04-27 16:54:40.000000 browser/7/
│ │ +drwxr-xr-x   0        0        0        0 2021-04-27 16:54:40.000000 browser/7/.brew/
│ │ +-rw-r--r--   0        0        0      421 2021-04-27 16:54:40.000000 browser/7/.brew/browser.rb
│ │ +drwxr-xr-x   0        0        0        0 2021-04-27 16:54:40.000000 browser/7/bin/
│ │ +-r-xr-xr-x   0        0        0      419 2021-04-27 16:54:40.000000 browser/7/bin/browser
```

**Comment by Bo98** [comment]
I wonder if it's a `Formula#source_modified_time` issue?

**Comment by Bo98** [comment]
Yes:

```sh
$ stat -f %Sm ~/Library/Caches/Homebrew/downloads/0d3dc7a1e45d7ac6eb8d853c8ecccad6a4ab6abbbf38038f552e4735ad15d3cc--hello-2.10.tar.gz
Nov 16 12:08:45 2014

$ stat -f %Sm ~/Library/Caches/Homebrew/downloads/e64b09988b7524a517b504c4f4bb47968f3251dbdaba439a33cb79d7011f9efa--browser
Apr 27 18:06:57 2021
```

**Comment by carlocab** [comment]
I guess being hosted in a Gist does that. Maybe we can move the source to formula-patches? 😄

**Comment by Bo98** [comment]
So we even have the correct mtime on GitHub tag archives?

**Comment by Bo98** [comment]
Seems like it also affects GitHub tag archives, but not uploaded release files. There's no `Last-Modified` header on tag archives or other raw files like gists.

What we should do here? Add some GitHub API calls to determine the tag/revision date?

**Comment by carlocab** [comment]
If we mirror those archives, can we guarantee a stable mtime? If we can that might be the way to go, as we have other uses for mirrors anyway.

**Comment by Bo98** [comment]
> If we mirror those archives, can we guarantee a stable mtime?

Only if the mirror is the primary source.

**Comment by carlocab** [comment]
I was thinking of a caching scheme, except the cache would be the mirror. Though I guess it's not easy to use a caching scheme where your cache object has a different checksum.

**Comment by Bo98** [comment]
There's also the problem that we need the `mtime` _before_ the build begins (since it'll affect `SOURCE_DATE_EPOCH`) but we typically only mirror on merge.

**Comment by Bo98** [comment]
Someone want to convince GitHub to add a `Last-Modified` header?

**Comment by MikeMcQuaid** [comment]
> If we mirror those archives, can we guarantee a stable mtime? If we can that might be the way to go, as we have other uses for mirrors anyway.

I'd rather we avoid mirroring anything that's `bottle :unneeded`. Note: we currently have no mirror host and I'm wanting to see how painful it is for us to not have one.

> So it is Gist & raw file specific because of there being no archive.

Note: all Gists can be cloned as Git repositories which have timestamps.



> Someone want to convince GitHub to add a `Last-Modified` header?

I don't think this is a likely medium-term solution 😁

**Comment by MikeMcQuaid** [comment]
> Note: all Gists can be cloned as Git repositories which have timestamps.

In this specific case: https://gist.github.com/318247.git will work. Might not be a bad idea to have an audit that prevents Gist URLs from being used for this reason.

File: Formula/browser.rb [hunk 1]
```diff
@@ -1,13 +1,12 @@
 class Browser < Formula
   desc "Pipe HTML to a browser"
   homepage "https://gist.github.com/318247/"
-  url "https://gist.github.com/defunkt/318247/raw/7720fc969d58a9bfdc74d71deaa15c1eb7582fc1/browser"
+  url "https://gist.github.com/318247.git",
+      revision: "21e65811a50b3cc8bb2b31c658279714657aab96"
   # This the gist revision number
   version "7"
   sha256 "273343d6cf9ed543ccea47b85a2ad2ef2afe7b7a2b956f2e7a24ce0175326dcc"
```
Comment (Bo98) [inline Formula/browser.rb:8]:
```suggestion
```

Comment (carlocab) [reply Formula/browser.rb:8]: Oops

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
@carlocab good to merge now?

**Comment by carlocab** [comment]
Yep, should be good:
```
❯ sha256sum *.tar.gz
6de7d877522742138bb0668ba627cdd166b2e0b1c698cc336ba704e68c6338e4  browser--7.arm64_big_sur.bottle.tar.gz
6de7d877522742138bb0668ba627cdd166b2e0b1c698cc336ba704e68c6338e4  browser--7.big_sur.bottle.tar.gz
6de7d877522742138bb0668ba627cdd166b2e0b1c698cc336ba704e68c6338e4  browser--7.catalina.bottle.tar.gz
6de7d877522742138bb0668ba627cdd166b2e0b1c698cc336ba704e68c6338e4  browser--7.mojave.bottle.tar.gz
```
Thanks @MikeMcQuaid!

**Pull request closed without merging**